File: backend/DTOs/LoginRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Axle.DTOs;

/// <summary>
/// Request model for user login.
/// </summary>
public class LoginRequest
{
    [Required]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;

    [Required]
    public string Password { get; set; } = string.Empty;

    public string? TwoFactorCode { get; set; }
}

File: frontend/lib/data/models/login_request.dart
/// Request model for user login.
///
/// Matches ASP.NET Core Identity login endpoint requirements.
class LoginRequest {
  const LoginRequest({
    required this.email,
    required this.password,
    this.twoFactorCode,
    this.twoFactorRecoveryCode,
  });

  final String email;
  final String password;
  final String? twoFactorCode;
  final String? twoFactorRecoveryCode;

  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'password': password,
      if (twoFactorCode != null) 'twoFactorCode': twoFactorCode,
      if (twoFactorRecoveryCode != null)
        'twoFactorRecoveryCode': twoFactorRecoveryCode,
    };
  }
}
